[Workouts page]

HugoGoHe opened the pull request:
Worked on the Workouts page where the user can create new workouts or access past workouts. Additionally, started working on the CurrentWorkout page, where the user will plan his workout. Until now, the user can create, delete or access past workouts and modify their name and date.

HugoGoHe committed:
New Post Button

- The button takes you to a page to see the preview of the image and then shows you your camera roll to select the photo, then it is shown on the preview page.

HugoGoHe committed:
Button Corrections

- Now the button does not show the preview first, but first lets the user select the image and then shows the preview.

- The image is selected and it is captured in a property of the class.

- The "previewSegue" is performed.

- In prepareForSegue the image is passed to the PostPreviewViewController.

- In the viewDidLoad of this class the UIImageView is initialized.

HugoGoHe committed:
Post Class

- Post.h
- Post.m

HugoGoHe committed:
didTapCheckmarck - PostPreviewViewController

- The button uploads the post to parse and sends the user to the first item of the tab bar.

- Does a fullscreen presentation

HugoGoHe committed:
User can post and collection view is populated

- The collection view shows 3 images per row without separation by margins.

- Now the back button works and sends the user to the WeightLoss page.

- The collectionview loads the images that have been posted by the current user.

HugoGoHe committed:
Bugs fixed

- The "undefined" bug in the author field was due to the fact that in some commit the author property of the post class had been deleted. Add it and assign it the value of currentUser in the .m file.

- Apparently the pull to refresh did not work because the collectionview did not occupy all the space of its container. I had to add the following lines:
        [self.collectionView addSubview:refreshControl];
        self.collectionView.alwaysBounceVertical = YES;

-Some posts were not being uploaded to the database. This was solved by closing my session and logging back in.

HugoGoHe committed:
Workouts page (not finished)

- Basic frontend of workouts page.
- Class of PFObject Workout created.
- Class for WorkoutCell created (TableViewCell)
- The tableView shows the workouts registered in the Parse's database.

HugoGoHe committed:
Creating and modifying workouts

- User can create a new workout.
- User can modify the name or date of an existing workout.
- Now the posts are presented from newest to oldest.
- Now the workouts are presented from most recent to oldest according to the date assigned by the user.

HugoGoHe committed:
Bugs

- Searchbar ends refreshing.
- When user taps into a cell, CurrentWorkoutViewController is loaded with that cell's data.

HugoGoHe committed:
[Deleting rows in workout tableView]

You can delete workouts from the table view by sliding the rows, the workout is deleted from the database and the tableView is reloaded after 0.2s

HugoGoHe committed:
[Deleting rows in a workout tableView] 

Now instead of using the timer, the object is deleted from the array and the deleteRowsAtIndexPaths method is used.

HugoGoHe committed:
Cleaning the code

cmnagarajan commented on Capstone Project/PostPreviewViewController.m:
```diff
@@ -0,0 +1,61 @@
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+    }
+    else {
+        NSLog(@"Camera 🚫 available so we will use photo library instead");
```
No need for this logging in the final checkin

cmnagarajan commented on Capstone Project/WeightLossViewController.m:
```diff
@@ -73,5 +95,70 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 }
 
+- (void) getPosts{
+    PFQuery *postQuery = [Post query];
```
Here PostQuery is actually a getQuery. The name of the class Post can be renamed to something else like WorkoutImage (if it makes sense). The names Post, Get and Put are restfull http APIs.

cmnagarajan commented on Capstone Project/WeightLossViewController.m:
```diff
@@ -73,5 +95,70 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+- (void) didPost{
+    
+    [self getPosts];
+    [self.collectionView reloadData];
```
There is no need for reloading it here again. Line 107 loads it. Reloading it here has no use

cmnagarajan commented on Capstone Project/WeightLossViewController.m:
```diff
@@ -73,5 +95,70 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+- (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    PictureGridCell *cell = [self.collectionView dequeueReusableCellWithReuseIdentifier:@"Picture Grid Cell" forIndexPath:indexPath];
+    
+    Post *post1 = self.arrayOfPosts[indexPath.row];
```
Name the variables meaningfully instead of post1

HugoGoHe committed:
[PR review comment corrections]

-  Deleted NSLog for checking if camera was working.
-  Changed the name of the class Post to ProgressPic
-  Changed the name of the variable postQuery to getQuery
- Deleted reloadData inside function didPost
- Changed the name of the variable post1 to progressPic
- Now fetching and reloading the Data inside completion blocks. 


Now the workouts table is updating itself without the need of pulling to refresh but the collectionView is still not doing that.
